spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py

```diff
@@ -26,41 +29,79 @@ class CellExplorerSortingExtractor(SortingExtractor):
-        assert session_info_matfile_path.is_file(), "No sessionInfo.mat file found in the folder!"
-        session_info_mat = loadmat(file_name=str(session_info_matfile_path.absolute()))
-        assert session_info_mat['sessionInfo']['rates'][0][0]['wideband'], "The sesssionInfo.mat file must contain " \
-            "a 'sessionInfo' struct with field 'rates' containing field 'wideband'!"
-        self._sampling_frequency = float(
-            session_info_mat['sessionInfo']['rates'][0][0]['wideband'][0][0][0][0]
-        )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
-
-        spikes_mat = loadmat(file_name=str(spikes_matfile_path.absolute()))
-        assert np.all(np.isin(['UID', 'times'], spikes_mat['spikes'].dtype.names)), \
-            "The spikes.cellinfo.mat file must contain a 'spikes' struct with fields 'UID' and 'times'!"
-
-        self._unit_ids = np.asarray(spikes_mat['spikes']['UID'][0][0][0], dtype=int)
+        assert (
+            spikes_matfile_path.is_file()
+        ), f"The spikes_matfile_path ({spikes_matfile_path}) must exist!"
+        
+        if sampling_frequency is None:
+            folder_path = spikes_matfile_path.parent
+            sorting_id = spikes_matfile_path.name.split(".")[0]
+            if session_info_matfile_path is None:
+                session_info_matfile_path = folder_path / f"{sorting_id}.sessionInfo.mat"
+
+            assert (
+                session_info_matfile_path.is_file()
+            ), f"No {sorting_id}.sessionInfo.mat file found in the folder!" 
+
+            try:
+                session_info_mat = scipy.io.loadmat(file_name=str(session_info_matfile_path))
+                self.read_session_info_with_scipy = True
+            except NotImplementedError:
+                session_info_mat = hdf5storage.loadmat(file_name=str(session_info_matfile_path))
+                self.read_session_info_with_scipy = False
+            
+            assert session_info_mat["sessionInfo"]["rates"][0][0]["wideband"], (
+                "The sesssionInfo.mat file must contain "
+                "a 'sessionInfo' struct with field 'rates' containing field 'wideband' to extract the sampling frequency!"
+            )
+            if self.read_session_info_with_scipy:
+                self._sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+            else:
+                self._sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+        else:
+            assert sampling_frequency is not None, "Could not load sampling frequency from files. Use the 'sampling_frequency' argument"
```

Comment:
@alejoe91  you suggested this assert but I am confused about it. The program only enters this else when the sampling_frequency is not None, so why check for it again?

Reply:
Right, you can remove this else ;)